ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala

```diff
@@ -235,75 +234,77 @@ private[indexer] class JdbcIndexer private[indexer] (
     new SubscriptionResourceOwner(readService)
 
   private def handleStateUpdate(
-      offset: Offset,
-      update: Update,
-  )(implicit loggingContext: LoggingContext): Future[Unit] = {
-    lastReceivedRecordTime = update.recordTime.toInstant.toEpochMilli
-
-    logger.trace(update.description)
-
-    metrics.daml.indexer.lastReceivedRecordTime.updateValue(lastReceivedRecordTime)
-    metrics.daml.indexer.lastReceivedOffset.updateValue(offset.toApiString)
-
-    val result = update match {
-      case PartyAddedToParticipant(
-          party,
-          displayName,
-          hostingParticipantId,
-          recordTime,
-          submissionId,
-          ) =>
-        val entry = PartyLedgerEntry.AllocationAccepted(
-          submissionId,
-          recordTime.toInstant,
-          domain.PartyDetails(party, Some(displayName), participantId == hostingParticipantId)
-        )
-        ledgerDao.storePartyEntry(offset, entry)
+      implicit loggingContext: LoggingContext): Flow[(Offset, Update), Unit, NotUsed] =
+    Flow[(Offset, Update)]
+      .wireTap(Sink.foreach[(Offset, Update)] {
+        case (offset, update) =>
+          lastReceivedRecordTime = update.recordTime.toInstant.toEpochMilli
```

Comment:
That mutable value is only used in this block (for metrics and logging). It might be concerning if this Sink starts to backpressure, but if it does, do we want to slow down processing downstream, for the sake of metrics, logging?

Reply:
I guess it's fine, thanks.

Reply:
Why are we using a mutable value at all? Is it for debugging? It looks to me like it's only used in the line that sets the metric value.

Couldn't we just use the following?
```
etrics.daml.indexer.lastReceivedRecordTime.updateValue(update.recordTime.toInstant.toEpochMilli)
```

Reply:
Good observation, @rautenrieth-da . Couldn't see the forest for the trees 😄 Removed this mutable var, seems it's not used anywhere else.